src/commands/createFunctionApp/stacks/FunctionAppEOLWarningStep.ts

```diff
@@ -0,0 +1,34 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.md in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+import { AzureWizardPromptStep, DialogResponses, openUrl } from "@microsoft/vscode-azext-utils";
+import { MessageItem } from "vscode";
+import { funcVersionLink } from "../../../FuncVersion";
+import { localize } from "../../../localize";
+import { getWorkspaceSetting, updateGlobalSetting } from "../../../vsCodeConfig/settings";
+import { IFunctionAppWizardContext } from "../IFunctionAppWizardContext";
+
+export class FunctionAppEOLWarningStep extends AzureWizardPromptStep<IFunctionAppWizardContext> {
+    public async prompt(context: IFunctionAppWizardContext): Promise<void> {
+        const settingKey: string = 'endOfLifeWarning';
+        if (getWorkspaceSetting<boolean>(settingKey)) {
+            const message = localize('endOfLife', "The chosen runtime stack has an end of support deadline coming up. " +
+                "After the deadline, function apps can be created and deployed, and existing apps continue to run. " +
+                "However, your apps won't be eligible for new features, security patches, performance optimizations, and support until you upgrade them.");
+            const continueOn: MessageItem = { title: localize('continueOn', 'Continue') };
+            let result: MessageItem = await context.ui.showWarningMessage(message, { modal: true }, DialogResponses.learnMore, continueOn, DialogResponses.dontWarnAgain,);
+            do {
```

Comment:
Using a do-while here makes it so that selecting items other than Learn More result in the learn more link opening. I think you'll want to use a normal while loop here.